frb_rust/src/macros.rs

```diff
@@ -0,0 +1,71 @@
+/// On WASM, [JsValue][wasm_bindgen::JsValue]s cannot be shared between scopes but instead can be
+/// ["transferred"]. Rust however is not aware of transferables and therefore cannot
+/// capture these values. This macro wraps a closure and returns a [TransferClosure] on WASM platforms
+/// which will capture these special values, or a normal [FnOnce] on other platforms.
+/// Note that the parameter names must match available variables/bindings from the outer scope.
+///
+/// ["transferred"]: https://developer.mozilla.org/en-US/docs/Glossary/Transferable_objects
+#[macro_export]
+macro_rules! transfer {
+    (|| $block:block) => {{
+        #[cfg(not(target_family = "wasm"))]
+        { move || $block }
+
+        #[cfg(target_family = "wasm")]
+        {
+            $crate::ffi::TransferClosure::new(vec![], move |_: &[JsValue]| $block)
```

Comment:
nit: Is it possible to merge this zero-arg case into the nonzero-arg case?

Reply:
Not really, since the zero-parameter case is considered a single `||` token which is not the same as two pipes.